posix: Attach a posix_spawn_disk_thread with glusterfs_ctx

mohit84 opened the pull request:
Currently posix xlator spawns posix_disk_space_threads per brick and in
case of brick_mux environment while glusterd attached bricks at maximum
level(250) with a single brick process in that case 250 threads are
spawned for all bricks and brick process memory size also increased.

Solution: Attach a posix_disk_space thread with glusterfs_ctx to
      spawn a thread per process basis instead of spawning a per brick

Fixes: #1482
Change-Id: I8dd88f252a950495b71742e2a7588bd5bb019ec7
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

mohit84 commented:
/run regression

mohit84 commented:
/run full regression

xhernandez commented:
Sorry, I did the review in #1558. Can you check the comments there and we'll continue here ?

mohit84 commented:
> Sorry, I did the review in #1558. Can you check the comments there and we'll continue here ?

sure i will resolve the comment and upload a new patch.

mohit84 committed:
posix: Attach a posix_spawn_disk_thread with glusterfs_ctx

Currently posix xlator spawns posix_disk_space_threads per brick and in
case of brick_mux environment while glusterd attached bricks at maximum
level(250) with a single brick process in that case 250 threads are
spawned for all bricks and brick process memory size also increased.

Solution: Attach a posix_disk_space thread with glusterfs_ctx to
          spawn a thread per process basis instead of spawning a per brick

Fixes: #1482
Change-Id: I8dd88f252a950495b71742e2a7588bd5bb019ec7
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>
